src/webpack.config.js

```diff
@@ -1,38 +1,78 @@
-import webpackCleanPlugin from 'clean-webpack-plugin'
+import fs from 'fs'
+import path from 'path'
+import webpack from 'webpack'
+import CleanPlugin from 'clean-webpack-plugin'
 
-export default (context) => {
-    const config = {
-      resolve: {
-        modulesDirectories: [context, `${context}/node_modules`]
-      },
-      entry: 'tapestry-wp/dist/client.js',
-      output: {
-        path: 'public',
-        filename: 'bundle.js'
-      },
-      module: {
-        loaders: [
-          {
-            test: /\.js$/,
-            exclude: /node_modules/,
-            loader: 'babel-loader',
-            query: {
-              presets: ['es2015', 'react']
-            }
-          }
-        ]
-      },
-      plugins: [
-        new webpackCleanPlugin(['public'], { root: context, verbose: false })
-        // new webpack.optimize.UglifyJsPlugin(),
-        // new webpack.optimize.OccurrenceOrderPlugin(),
-        // new webpack.optimize.DedupePlugin(),
-        // new webpack.DefinePlugin({
-        //   'process.env': {
-        //     'NODE_ENV': JSON.stringify('production')
-        //   }
-        // })
-      ]
+const env = process.env.NODE_ENV
+
+
+export default (ctx, isTree = false) => {
+
+  const config = {
+    resolve: {
+      modules: [ctx, `${ctx}/node_modules`]
+    },
+    module: {
+      rules: [{
+        test: /\.js$/,
+        exclude: /node_modules/,
+        loader: 'babel-loader',
+        options: {
+          presets: ['es2015', 'react']
```

Comment:
Yeah I've had a few plays with this, I'll try again on a separate branch, maybe we can pair-program it next week?